doc/source/dev/howto_build_docs.rst

```diff
@@ -53,7 +53,13 @@ Dependencies
 All of the necessary dependencies for building the NumPy docs except for
 Doxygen_ can be installed with::
 
-    pip install -r requirements/doc_requirements.txt
+    pip install  --pre --force-reinstall   --extra-index-url \
+    https://pypi.anaconda.org/scientific-python-nightly-wheels/simple \
+    -r requirements/doc_requirements.txt
+
+The extra parameters of command above is to tell pip install nightly
+build packages for dependency. For MacOS arm platform, the command above may
+not install needed packages so the NumPy docs can't be built.
```

Comment:
Maybe we can move this to a separate `.. note::` (or similar admonition) instead of replacing the existing instructions. This extra stuff is likely only to be necessary during the NumPy 2.0 transition period. Once 2.0 is out and the major doc dependencies have made releases that support 2.0 this advice will (should) no longer be necessary and removed again.